api/lib/domain/events/handle-badge-acquisition.js

```diff
@@ -0,0 +1,40 @@
+const handleBadgeAcquisition = async function({
+  domainTransaction,
+  assessmentCompletedEvent,
+  badgeCriteriaService,
+  badgeAcquisitionRepository,
+  badgeRepository,
+  campaignParticipationResultRepository,
+}) {
+  if (completedAssessmentBelongsToACampaign(assessmentCompletedEvent)) {
+    const badge = await fetchPossibleCampaignAssociatedBadge(assessmentCompletedEvent, badgeRepository);
+    if (isABadgeAssociatedToCampaign(badge)) {
+      const campaignParticipationResult = await fetchCampaignParticipationResults(assessmentCompletedEvent, campaignParticipationResultRepository);
+      if (isBadgeAcquired(campaignParticipationResult, badgeCriteriaService)) {
+        await badgeAcquisitionRepository.create(domainTransaction, { badgeId: badge.id, userId: assessmentCompletedEvent.userId });
+      }
+    }
+  }
+};
+
+function completedAssessmentBelongsToACampaign(assessmentCompletedEvent) {
+  return assessmentCompletedEvent.targetProfileId != null;
+}
+
+async function fetchPossibleCampaignAssociatedBadge(assessmentCompletedEvent, badgeRepository) {
```

Comment:
Est-ce qu'ici on pourrait imaginer un handler qui soit un objet, avec cette fonction qui serait une méthode de l'objet, et qui passerait par `this` pour appeler ses dépendances ? Ça nécessiterait un mécanisme d'injection un peu différent : à expérimenter plus tard.